backend/pyproject.toml

```diff
@@ -1,55 +1,64 @@
-[tool.poetry]
+# Once uv supports setting dynamic versions, replace the version line with:
+# dynamic = ["version"]
+
+[project]
 name = "monarch-py"
 version = "0.0.0"
 description = "Python package for interacting with Monarch Initiative knowledge graph"
 authors = [
-    "glass-ships <glass.ships@outlook.com>",
-    "kevin schaper <kevin@tislab.org>",
-    "The Monarch Initiative <info@monarchinitiative.org>",
+    { name = "The Monarch Initiative", email = "info@monarchinitiative.org" },
+    { name = "kevin schaper", email = "kevin@tislab.org" },
+    { name = "glass-ships", email = "glass.ships@outlook.com" },
 ]
 
+requires-python = ">=3.10, <4.0"
+
+dependencies = [
+    "beautifulsoup4>=4.13.4",
+    "bioregistry~=0.10.57",
+    "curies<1",
+    "docker>=7.1.0",
+    "fastapi>=0.115.12,<1",
+    "gunicorn>=23.0.0",
+    "linkml==1.8.3",
+    "loguru",
+    "oaklib>=0.6.6",
+    "prefixmaps==0.2.4",
+    "pydantic>=2,<3",
+    "pystow>=0.5.4",
+    "requests>=2.31.0",
+    "rich",
+    "spacy>=3.8.7",
+    "typer>=0.12.0",
+]
 
-packages = [{ include = "monarch_py", from = "src" }]
-
-
-[tool.poetry.dependencies]
-### Core dependencies
-python = "^3.9"
-pydantic = "^2"
-curies = "<1"
-linkml = "1.8.3"
-prefixmaps = "^0.2"
-bioregistry = "^0.10.57"
-oaklib = ">=0.6.6"
-
-docker = "^7.1.0"
-fastapi = "^0.103.1"
-gunicorn = "^22.0.0"
-loguru = "*"
-pystow = ">=0.5.4"
-requests = "^2.31.0"
-rich = "*"
-typer = "^0.12.0"
-beautifulsoup4 = "^4.12.3"
-spacy = "^3.7.6"
+[project.optional-dependencies]
+dev = [
+    "pytest>=8.2.0",
+    "mkdocs>=1.6.0",
+    "mkdocs-material>=9.5.23",
+    "mkdocstrings[python]>=0.29.1",
+    "black>=24.4.2",
+    "ruff",
+    "uvicorn[standard]>=0.29.0",
+    "httpx>=0.27",
+    "scholarly",
+    "coverage>=7.5.1",
+]
 
-[tool.poetry.group.dev.dependencies]
-pytest = "^8.2.0"
-mkdocs = ">=1.6.0"
-mkdocs-material = ">=9.5.23"
-mkdocstrings = { version = ">=0.25.1", extras = ["python"] }
-black = "^24.4.2"
-ruff = "*"
-uvicorn = { extras = ["standard"], version = "^0.29.0" }
-httpx = "^0.27"
-scholarly = "*"
-coverage = "^7.5.1"
+[project.scripts]
+monarch = "monarch_py.cli:app"
+monarch-api = "monarch_py.api.main:run"
 
+[build-system]
+requires = ["hatchling", "uv-dynamic-versioning"]
+build-backend = "hatchling.build"
 
-[tool.poetry.scripts]
-monarch = "monarch_py.cli:app"
-monarch-api = { callable = "monarch_py.api.main:run" }
+[tool.hatch.version]
+source = "uv-dynamic-versioning"
```

Comment:
under this i think its standard to also include, just as standard configuration for it and we might want to add a fallback version

```
[tool.uv-dynamic-versioning]
vcs = "git"
style = "pep440"
fallback-version = "0.0.0"
```